sync/class.jetpack-sync-module-users.php

```diff
@@ -54,6 +62,42 @@ public function init_listeners( $callable ) {
 		add_action( 'wp_masterbar_logout', $callable, 10, 0 );
 	}
 
+	public function delay_if_within_wp_insert_user() {
+		if ( ! Jetpack::is_function_in_backtrace( 'do_delayed_calls' ) ) {
+			if ( Jetpack::is_function_in_backtrace( 'wp_insert_user' ) ) {
+				self::$calls_within_wp_insert_user[ current_action() ][] = func_get_args();
+
+				return;
+			}
+		}
+		call_user_func_array( self::$callable, func_get_args() );
+	}
+
+	public function do_delayed_calls() {
+		// We want the save user action to be synced first
+		if ( isset( self::$calls_within_wp_insert_user['jetpack_sync_save_user'] ) ) {
+			$jetpack_sync_save_user = self::$calls_within_wp_insert_user['jetpack_sync_save_user'];
+			unset( self::$calls_within_wp_insert_user['jetpack_sync_save_user'] );
+			self::$calls_within_wp_insert_user = array_merge( array( 'jetpack_sync_save_user' => $jetpack_sync_save_user), self::$calls_within_wp_insert_user );
```

Comment:
Consider adding a prepend_to_array() function so you can call 
```prepend_to_array(array( 'jetpack_sync_save_user' => $jetpack_sync_save_user), self::$calls_within_wp_insert_user )```
and it will be explicit that you are adding something to the beginning of the array, rather than implicit and requiring background knowledge of the objective.